web/vite.config.js

```diff
@@ -12,6 +13,20 @@ const upstream = {
 };
 
 export default defineConfig({
+  build: {
+    // largest chunk size is maplibre-gl
+    chunkSizeWarningLimit: 800,
+    rollupOptions: {
+      output: {
+        // optimize chunks size https://github.com/vitejs/vite/discussions/9440
```

Comment:
I think this is going to create tons of small chunks which isn't really better either. Vite's defaults are pretty decent and so this feels unlikely to be an overall improvement

What I was thinking would be ideal would be if we could use a dynamic import to avoid putting some of the larger libraries in the main chunk. E.g. if I recall correctly, I think the map viewer is quite large and not loaded until you click into an image. Currently I believe it gets loaded straightaway and we pay the cost for it even if the user doesn't use it. If we do a dynamic import, we can wait to load it so that it doesn't slow down loading of the JavaScript needed for initial page load. E.g. maybe we could load it in `onMount` so that it doesn't block the initial JS loading, but is there in time for the user to use it

Reply:
Here's a PR to split the big chunk into two: https://github.com/immich-app/immich/pull/7057

They're both over the limit, so doesn't help in terms of removing the warning, but it should be a better user experience

I don't think there's a way to really shrink the photo sphere chunk further, so I guess we could set the limit at 600 kB to silence the warning for that one. We can probably do more with the asset viewer by splitting out the map stuff as well

Reply:
Oh that's nice 😃

Yeah, I saw another project doing the same thing because of maplibre. Sounds reasonable.